libs/ledgerjs/packages/hw-app-eth/src/modules/EIP712/index.ts

```diff
@@ -67,12 +79,23 @@ const makeRecursiveFieldStructImplem = (
         if (fieldType) {
           await recursiveFieldStructImplem(
             destructTypeFromString(fieldType),
-            fieldValue
+            fieldValue,
+            `${path}.${fieldName}`
           );
         }
       }
     } else {
-      await EIP712SendStructImplem(transport, {
+      if (filters) {
```

Comment:
can it be simplified with 
```
const filter = filters?.fields.find((f) => path === f.path);
if (filter)
```

I'm not 100% sure

Reply:
Yup, good catch ! I'm updating that 👌